graylog2-server/src/main/java/org/graylog2/indexer/counts/Counts.java

```diff
@@ -36,22 +38,21 @@ public Counts(Client client, IndexSetRegistry indexSetRegistry) {
     }
 
     public long total() {
-        final SearchRequest request = c.prepareSearch(indexSetRegistry.getManagedIndicesNames())
-                .setSize(0)
-                .request();
-        return c.search(request).actionGet().getHits().totalHits();
+        return totalCount(indexSetRegistry.getManagedIndicesNames());
     }
 
     public long total(final IndexSet indexSet) {
-        final String[] names = indexSet.getManagedIndicesNames();
+        return totalCount(indexSet.getManagedIndicesNames());
+    }
 
+    private long totalCount(final String[] indexNames) {
         // Return 0 if there are no indices in the given index set. If we run the query with an empty index list,
         // Elasticsearch will count all documents in all indices and thus return a wrong count.
-        if (names.length == 0) {
+        if (requireNonNull(indexNames).length == 0) {
```

Comment:
Why the `requireNonNull()` here?

If `indexNames` is `null`, it will throw an NPE anyway.